app/helpers/application_helper.rb

```diff
@@ -1,2 +1,15 @@
 module ApplicationHelper
+  def current_cart
+    @cache_key = current_user.carts.last&.cache_key_with_version
+    if current_user.nil?   
```

Comment:
Let's split this condition for user out of this statement, something like this:

```ruby
def current_cat
  @cache_key = ...
  redirect_to new_user_session_path if current_user.nil?

  # avoid using double negation in ruby
# !Rails.cache.read(@cache_key).nil? replace with Rails.cache.read(@cache_key).present?
# so it will continue like this: 
if Rails.cache.read(@cache_key).present?
 # do something
else
  # do something
end
@cart

end

end
```

Reply:
Split and changed